tools/@aws-cdk/cfn2ts/lib/index.ts

```diff
@@ -27,82 +32,113 @@ export default async function(scopes: string | string[], outPath: string, option
     const generator = new CodeGenerator(name, spec, affix, options);
     generator.emitCode();
     await generator.save(outPath);
+    outputFiles.push(generator.outputFile);
 
-    const augs = new AugmentationGenerator(name, spec, affix);
+    const augs = new AugmentationGenerator(name, spec, affix, options);
     if (augs.emitCode()) {
       await augs.save(outPath);
+      outputFiles.push(augs.outputFile);
     }
 
     const canned = new CannedMetricsGenerator(name, scope);
     if (canned.generate()) {
       await canned.save(outPath);
+      outputFiles.push(canned.outputFile);
     }
   }
-}
-
-export async function generateAll(outPath: string, options: CodeGeneratorOptions): Promise<pkglint.ModuleDefinition[]> {
-  const scopes = cfnSpec.namespaces();
-  const modules = new Array<pkglint.ModuleDefinition>();
 
-  for (const scope of scopes) {
-    const spec = cfnSpec.filteredSpecification(s => s.startsWith(`${scope}::`));
-    const module = pkglint.createModuleDefinitionFromCfnNamespace(scope);
-    const packagePath = path.join(outPath, module.moduleName);
-    const libPath = path.join(packagePath, 'lib');
+  return outputFiles;
+}
 
-    modules.push(module);
+export interface GeneratorOptions extends CodeGeneratorOptions, AugmentationsGeneratorOptions {
 
-    if (Object.keys(spec.ResourceTypes).length === 0) {
-      throw new Error(`No resource was found for scope ${scope}`);
-    }
-    const name = packageName(scope);
-    const affix = computeAffix(scope, scopes);
+  /**
+    * Map of CFN Scopes to modules
+    */
+  scopeMap: Record<string, string[]>;
```

Comment:
Using this type and the ModuleMap interface to represent almost the same thing I found really confusing. I think we should encapsulate the process of turning the json file into the `ModuleMap` type. It's a bit hard to explain in comments, so I'm going to push to the branch to describe what I mean.

Reply:
See the commit here: https://github.com/aws/aws-cdk/pull/24263/commits/d91b82134ed5b3759d3491f3a0d7dbe785f68d9c